docs/pow.md

```diff
@@ -0,0 +1,71 @@
+# Proof-of-work in the faucet
+
+The faucet uses POW challenges for anti-dos and dynamically shifts the
+difficulty based on the faucet's balance.
+
+Generally, the less bitcoin in the faucet, the more difficult it is to get out
+via solving the POW challenge.
+
+This was initially implemented via a logarithmic based function but since the
+difficulty parameter already exponentially increases the difficulty, we switched
+to a linear approach which is easier to read and is less likely to blow up.
```

Comment:
Why this is important to mention here? Just focus on the solution at hand.
For what's worth, this kinda of backlog details belongs to a `CHANGELOG.md`.